### ISSUE TITLE
Support dynamic learning rate schedules

### ISSUE THREAD

**Issue by mdekstrand** [lenskit/lenskit#622]
We would like to support dynamic learning rate schedules in FunkSVD and other iterative algorithms, instead of the fixed learning rates that we currently use.

In order to do this, we need:
- [ ] An interface representing a learning rate schedule (`LearningRateSchedule`, in the `org.grouplens.lenskit.iterative` package)
- [ ] Constent schedule implementation
- [ ] Exponential decay schedule (using a standard schedule)
- Use the learning rate schedule in several places:
  - [ ] Gradient descent least-squares builder
  - [ ] FunkSVD model builder
  - [ ] FunkSVD run-time training
  - [ ] OrdRec
